src/platform/packages/shared/kbn-data-grid-in-table-search/README.md

```diff
@@ -0,0 +1,21 @@
+# @kbn/data-grid-in-table-search
+
+This package allows to extend `EuiDataGrid` with in-table search.
+
+If you are already using `UnifiedDataTable` component, you can enable in-table search simply by passing `enableInTableSearch` prop to it.
+
+If you are using `EuiDataGrid` directly, you can enable in-table search by importing this package and following these steps:
+
+1. include `useDataGridInTableSearch` hook in your component 
+2. pass `inTableSearchControl` to `EuiDataGrid` inside `additionalControls.right` prop or `renderCustomToolbar`
+3. pass `inTableSearchCss` to the grid container element as `css` prop
+4. update your `renderCellValue` to accept `inTableSearchTerm` and `onHighlightsCountFound` props
+5. update your `renderCellValue` to wrap the cell content with `InTableSearchHighlightsWrapper` component.
+
+Examples could be found inside `kbn-unified-data-table` package or in this [tmp commit](https://github.com/elastic/kibana/pull/206454/commits/2c56de09341ee9ce20c0f78ee6a89da0347014f0).
```

Comment:
I experimented with enabling the search for Lens table. The mentioned commit points to the changes for it.
The question would be where to add such search input around the Lens table. If it's just above the table then it creates too much empty space.

![lens-search](https://github.com/user-attachments/assets/a4c7cc50-8978-44aa-ba1f-decdf0cc29e2)